### PR TITLE
Development environment in Docker

### PR DESCRIPTION
**Pull request by ne-sachirou**
We need Vagrant for the future operation confirmation, so I don't remove it.

The syncronize system is based on https://speakerdeck.com/ne_sachirou/elixir-on-containers?slide=14 .

```console
% make help
clean           	 Clean all resources.
release-build   	 Build mackerel_agent for release.
resync          	 Re-sync the Docker volume when it's broken.
sh              	 Start a development shell.
start           	 Build and start mackerel_agent in debug mode.
test            	 Test.
```

```console
% make sh
docker-compose pull
Pulling app-src ... done
Pulling app     ... done
WARNING: Some service image(s) must be built from source by running:
    docker-compose build app
docker-compose build --build-arg BUILDKIT_INLINE_CACHE=1 --force-rm --pull
app-src uses an image, skipping
WARNING: Native build is an experimental feature and could change at any time
Building app
[+] Building 0.8s (11/11) FINISHED                                                                                                                                                                                          
 => [internal] load .dockerignore                                                                                                                                                                                      0.0s
 => => transferring context: 34B                                                                                                                                                                                       0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                   0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/rust:1-slim                                                                                                                                                         0.8s
 => [1/5] FROM docker.io/library/rust:1-slim@sha256:a99bc59105d13bef9d3028be8de9be157232bab6998b10d900a0c20e32939650                                                                                                   0.0s
 => [internal] load build context                                                                                                                                                                                      0.0s
 => => transferring context: 96B                                                                                                                                                                                       0.0s
 => CACHED [2/5] WORKDIR /mnt                                                                                                                                                                                          0.0s
 => CACHED [3/5] RUN apt-get -y update  && apt-get install -y --no-install-recommends     build-essential     curl     git     libssl-dev     pkg-config     rsync  && rustup default nightly  && cargo install cargo  0.0s
 => CACHED [4/5] COPY Cargo.toml Cargo.lock rust-toolchain ./                                                                                                                                                          0.0s
 => CACHED [5/5] RUN mkdir -p src  && touch src/lib.rs  && cargo build  && mv target /tmp                                                                                                                              0.0s
 => exporting to image                                                                                                                                                                                                 0.0s
 => => exporting layers                                                                                                                                                                                                0.0s
 => => writing image sha256:067dd5c8302d0cbc9fcd0b8fad312d776cc4f872ae86d9a5603754465df8595f                                                                                                                           0.0s
 => => naming to docker.io/library/mackerel-agent-rs:latest                                                                                                                                                            0.0s
 => exporting cache                                                                                                                                                                                                    0.0s
 => => preparing build cache for export                                                                                                                                                                                0.0s
Successfully built 067dd5c8302d0cbc9fcd0b8fad312d776cc4f872ae86d9a5603754465df8595f
docker-compose run --rm app bash
WARNING: Native build is an experimental feature and could change at any time
Creating network "mackerel-agent-rs_default" with the default driver
Creating mackerel-agent-rs_app-src_1 ... done
Creating mackerel-agent-rs_app_run   ... done
+ '[' '!' -d node_modules ']'
+ rsync -au /tmp/target .
+ set +u
+ '[' -z bash ']'
+ set -u
+ exec bash

# make start
cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.31s
     Running `target/debug/mackerel_agent`
[src/main.rs:56] Config::from_file(path) = Config {
    apibase: "https://api.mackerelio.com/",
    apikey: "*****",
    roles: Some(
        [
            "*****",
        ],
    ),
}
```

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  app-src:
+    image: eugenmayer/unison:2.51.2.2
+    environment:
+      APP_VOLUME: /app_sync
+      HOST_VOLUME: /host_sync
+      TZ: Asia/Tokyo
+      UNISON_ARGS: |
+        -debug default -prefer /host_sync -numericids -auto -batch -fat \
+        -ignore 'Name .#*' \
+        -ignore 'Name target/*' \
```
Comment (ne-sachirou) [inline docker-compose.yml:30]: This makes separate in-Docker environment from the host one.

File: Vagrantfile [hunk 2]
```diff
@@ -2,18 +2,18 @@
 # vi: set ft=ruby :
 
 Vagrant.configure(2) do |config|
-  config.vm.box = "debian/buster64"
-  config.vm.box_version = "10.3.0"
-  config.vm.provider "virtualbox" do |vm|
+  config.vm.box = 'debian/buster64'
```
Comment (ne-sachirou) [inline Vagrantfile:5]: `rubocop -a Vagrantfile`

**Comment by Krout0n** [comment]
@ne-sachirou no revising to README?

**Comment by ne-sachirou** [comment]
OK, I will.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[Makefile]
+indent_style = tab
+
+[*.rs]
```
Comment (Krout0n) [inline .editorconfig:14]:
```suggestion
# TODO: Force us to apply `$ cargo fmt`, then we won't need a config for Rust
[*.rs]
```

**Review by Krout0n — COMMENTED** [review]
I have no sense of dev with Docker, can you get to be main-reviewer instead of me? @dekokun

**Review by dekokun — APPROVED** [review]
LGTM !!

**Pull request merged**